### PR TITLE
Add etcd module (#4970)

### PR DESCRIPTION
**Pull request by berfinsari**
The Etcd module uses [Etcd v2 API](https://coreos.com/etcd/docs/latest/v2/api.html) to collect metrics.

### CODE REVIEW

**Comment by berfinsari** [comment]
#4970

**Comment by berfinsari** [comment]
I run make fmt command to cleanup the styles but I still get the same error. Is there anything that I can do for this unsuccessful check?

File: metricbeat/_meta/beat.reference.yml [hunk 1]
```diff
@@ -0,0 +1,452 @@
+########################## Metricbeat Configuration ###########################
```
Comment (ruflin) [inline metricbeat/_meta/beat.reference.yml]: I think this file was added by accident?

File: metricbeat/_meta/fields.generated.yml [hunk 2]
```diff
@@ -0,0 +1,6769 @@
+- key: common
```
Comment (ruflin) [inline metricbeat/_meta/fields.generated.yml]: This file should not be checked in.

**Comment by exekias** [comment]
Hi @berfinsari, thank you for contributing! As @ruflin pointed out, there is an issue with the testing container, you will need to add a valid HEALTHCHECK, for some reason the current one is failing, so we get a timeout waiting for the service to be ready to run tests

File: metricbeat/module/etcd/_meta/config.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+- module: etcd
+  metricsets: ["leader", "self", "store"]
+  enabled: false
```
Comment (exekias) [inline metricbeat/module/etcd/_meta/config.yml]: Can you drop the `enabled` field? We leave it to the default now

**Comment by tsg** [comment]
jenkins, test it

**Comment by exekias** [comment]
I'll have a look to the healthcheck

**Comment by ruflin** [comment]
@exekias Did you have a chance to look into the healthcheck?

@berfinsari Could you rebase and squash your PR on top of master. There seems to be 2 conflicts.

**Comment by exekias** [comment]
I've fixed conflicts and health check (`curl` was not available on that image so I switched to `wget`), let see if tests pass this time

**Comment by exekias** [comment]
jenkins test it please

**Comment by exekias** [comment]
jenkins test it, please

**Comment by exekias** [comment]
I fixed the container health issue, but tests seem to be broken, @berfinsari do you want to have a look? I can take it tomorrow if not

**Comment by berfinsari** [comment]
@exekias Thanks for the healthcheck, i check the integration tests

**Comment by ruflin** [comment]
jenkins, test it

File: metricbeat/module/etcd/_meta/env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ETCD_HOST=localhost
```
Comment (exekias) [inline metricbeat/module/etcd/_meta/env]: I think you want to put `ETCD_HOST=etcd` here, it will probably fix tests

**Comment by exekias** [comment]
jenkins test it please

**Comment by exekias** [comment]
Finally tests are green :green_heart:, thank you for your contribution!!

**Pull request merged**